ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/MsckDesc.java

```diff
@@ -36,19 +34,16 @@
   private static final long serialVersionUID = 1L;
 
   private final String tableName;
-  private final ArrayList<LinkedHashMap<String, String>> partitionsSpecs;
+  private final List<Map<String, String>> partitionsSpecs;
   private final String resFile;
   private final boolean repairPartitions;
   private final boolean addPartitions;
   private final boolean dropPartitions;
 
-  public MsckDesc(String tableName, List<? extends Map<String, String>> partitionSpecs, Path resFile,
+  public MsckDesc(String tableName, List<Map<String, String>> partitionsSpecs, Path resFile,
       boolean repairPartitions, boolean addPartitions, boolean dropPartitions) {
     this.tableName = tableName;
-    this.partitionsSpecs = new ArrayList<LinkedHashMap<String, String>>(partitionSpecs.size());
```

Comment:
earlier; the approach was to make a new copy

Reply:
No need for a new copy.